centreon/src/CentreonLegacy/Core/Module/Upgrader.php

```diff
@@ -71,9 +83,24 @@ public function upgrade()
         // finally, upgrade to current version
         $this->upgradeVersion($this->moduleConfiguration['mod_release']);
 
+        // Clearing symfony cache to properly enable API routes.
+        $this->clearCache();
+
         return $this->moduleId;
     }
 
+    private function clearCache(): void
+    {
+        $process = new Process(['php', 'bin/console', 'cache:clear']);
+        $process->setWorkingDirectory(_CENTREON_PATH_);
+
+        $process->run();
+
+        if (! $process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+    }
+
```

Comment:
IIRC @marlaud-centreon suggested to only clear the routes related cache, not the whole thing.

Something like this was suggested in the related thread:

```php
foreach (glob("$cacheDir/url_*.php") as $file) {
    @unlink($file);
}
```

Reply:
👍